src/dr/evomodel/tree/TreeModel.java

```diff
@@ -262,7 +262,10 @@ public boolean hasNodeHeights() {
     public boolean hasBranchLengths() {
         return true;
     }
-
+    abstract public boolean hasNodeTraits();
+    abstract public void swapAllTraits(NodeRef node1, NodeRef node2);
+    abstract public boolean hasRates();
+    
```

Comment:
Lifting these to the parent class to make it is easier to BigFastTrees and DefaultTrees interchangeably.